doc/api/n-api.md

```diff
@@ -0,0 +1,3150 @@
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API increments the reference count for the reference
+passed in and returns the resulting reference count.
+
+
+#### napi_reference_unref
+##### Signature
+```C
+NODE_EXTERN napi_status napi_reference_unref(napi_env e,
+                                             napi_ref ref,
+                                             int* result);
+```
+##### Parameters
+- `[in] env `: The environment that the API is invoked under
+- `[in] ref`: `napi_ref` to which we'll decrement the reference count
+- `[out] result`: the new reference count
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API decrements the reference count for the reference
+passed in and returns the resulting reference count.
+
+
+#### napi_get_reference_value
+##### Signature
+```C
+NODE_EXTERN napi_status napi_get_reference_value(napi_env e,
+                                                 napi_ref ref,
+                                                 napi_value* result);
+```
+
+the `napi_value passed` in or out of these methods is a handle to the
+object to which the reference is related.
+##### Parameters
+- `[in] env `: The environment that the API is invoked under
+- `[in] ref`: `napi_ref` for which we requesting the corresponding Object
+- `[out] result`: the `napi_value` for the Object referenced by the
+`napi_ref`
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+If still valid, this API returns the `napi_value` representing the
+JavaScript Object associated with the `napi_ref`. Otherise, result
+will be NULL.
+
+## Module registration
+
+N-API modules are registered in the same manner as other modules
+except that instead of using the `NODE_MODULE` macro you use:
+
+```C
+NAPI_MODULE(addon, Init)
+```
+
+The next difference is the signature for the `Init` method. For a N-API
+module it is as follows:
+
+```C
+void Init(napi_env env, napi_value exports, napi_value module, void* priv);
+```
+
+As with any other module, functions are exported by either adding them to
+the `exports` or `module` objects passed to the `Init` method.
+
+For example, to add the method `hello` as a function so that it can be called
+as a method provided by the addon:
+
+```C
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_status status;
+  napi_property_descriptor desc =
+    {"hello", Method, 0, 0, 0, napi_default, 0};
+  status = napi_define_properties(env, exports, 1, &desc);
+}
+```
+
+For example, to set a function to be returned by the `require()` for the addon:
+
+```C
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_status status;
+  napi_property_descriptor desc =
+    {"exports", Method, 0, 0, 0, napi_default, 0};
+  status = napi_define_properties(env, module, 1, &desc);
+}
+```
+
+For example, to define a class so that new instances can be created
+(often used with [object wrap]()):
```

Comment:
object wrap? This should probably be stylized, or written as a full noun like `object wrapping`?